components/director/internal/repo/delete.go

```diff
@@ -0,0 +1,41 @@
+package repo
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kyma-incubator/compass/components/director/internal/persistence"
+	"github.com/pkg/errors"
+)
+
+type Deleter struct {
+	tableName    string
+	idColumn     string
+	tenantColumn string
+}
+
+func NewDeleter(tableName, tenantColumn, idColumn string) *Deleter {
+	return &Deleter{tableName: tableName, idColumn: idColumn, tenantColumn: tenantColumn}
+}
+
+func (g *Deleter) Delete(ctx context.Context, tenant string, id string) error {
+	persist, err := persistence.FromCtx(ctx)
+	if err != nil {
+		return err
+	}
+
+	q := fmt.Sprintf("DELETE FROM %s WHERE %s=$1 AND %s=$2", g.tableName, g.tenantColumn, g.idColumn)
```

Comment:
In `labels` domain we have 3 parameters needed for delete. Please use variadic parameter, to allow using generic repository in more cases. Please think about variadic parameters or `map[string]string` of required conditions.